### PR TITLE
Allow to specify the ToC depth of the sidebar

### LINKED ISSUE #732: ToC depth of sidebar
Is currently capped to h2 headers. 

We are having h2 - h4 headers and missing the ToC in the sidebar completely: https://mlr-org.github.io/mlr/articles/tutorial/devel/feature_selection.html, https://raw.githubusercontent.com/mlr-org/mlr/master/vignettes/tutorial/devel/feature_selection.Rmd

Could the ToC depth become an argument to `build_articles()`?

crossref https://github.com/mlr-org/mlr/issues/2236

### PR DESCRIPTION
**Pull request by pat-s**
fixes #732 

relates to https://github.com/mlr-org/mlr/issues/2236

### CODE REVIEW

**Comment by pat-s** [comment]
@hadley @jayhesselberth 
Any feedback on this? :)

File: R/build.r [hunk 1]
```diff
@@ -353,8 +358,8 @@ build_site_local <- function(pkg = ".",
     override = override,
     preview = FALSE
   )
-  build_articles(pkg, lazy = lazy, override = override, preview = FALSE)
-  build_tutorials(pkg, override = override, preview = FALSE)
+  build_articles(pkg, lazy = lazy, override = override, preview = FALSE,
+                 toc_depth = toc_depth)
```
Comment (jayhesselberth) [inline R/build.r]: Did you mean for the `build_tutorials` call to disappear?

Comment (pat-s) [reply R/build.r]: Ah no, ofc not, this was by accident. Thanks for catching this!

**Comment by jayhesselberth** [comment]
LGTM. I will let @hadley comment too.

**Comment by hadley** [comment]
Wouldn't it be better to specify this in the yaml?

**Pull request closed without merging**